src/gui/torrentoptionsdialog.cpp

```diff
@@ -130,8 +156,31 @@ TorrentOptionsDialog::TorrentOptionsDialog(QWidget *parent, const QVector<BitTor
                 allSameLSD = false;
             }
         }
+        if (allSameSequential)
+        {
+            if (torrent->isSequentialDownload() != isFirstTorrentLSequentialEnabled)
+            {
+                m_ui->checkSequential->setCheckState(Qt::PartiallyChecked);
+                allSameSequential = false;
+            }
+        }
+        if (allSameFirstLastPieces)
+        {
+            if (torrent->hasFirstLastPiecePriority() != isFirstTorrentFirstLastPiecesEnabled)
+            {
+                m_ui->checkFirstLastPieces->setCheckState(Qt::PartiallyChecked);
+                allSameFirstLastPieces = false;
+            }
+        }
     }
 
+    if (allSameSavePath)
+        m_ui->savePath->setSelectedPath(firstTorrentSavePath);
+    if (m_ui->checkAutoTMM->checkState() != Qt::PartiallyChecked)
```

Comment:
The following will look more consistent:

```suggestion
    if (allSameAutoTMM)
```

I am also sure that you should not change state of `m_ui->checkAutoTMM` until all the torrents are checked.

Reply:
>I am also sure that you should not change state of `m_ui->checkAutoTMM` until all the torrents are checked.

The same is about other similar checkboxes. You should set these states only after all torrents are traversed:
```c++
if (allSameAutoTMM)
    m_ui->checkAutoTMM->setChecked(isFirstTorrentAutoTMMEnabled);
else
    m_ui->checkAutoTMM->setCheckState(Qt::PartiallyChecked);
```